lightning/src/ln/outbound_payment.rs

```diff
@@ -763,7 +763,7 @@ impl OutboundPayments {
 				// initial HTLC-Add messages yet.
 			},
 			PaymentSendFailure::PathParameterError(results) => {
-				Self::push_payment_path_failed_evs(payment_id, payment_hash, None, route.paths, results.into_iter(), pending_events);
+				Self::push_path_failed_evs_and_scids(payment_id, payment_hash, &mut route_params, route.paths, results.into_iter(), pending_events);
```

Comment:
Since this is a `PathParameterError`, IIUC we shouldn't considered the channel to have failed.

Reply:
Prefer to go back to an `Option`? We push the scid here but it's never actually *used* since there's no retry

Reply:
Oh, I see what you're saying. Good point, I'll fix

Reply:
Is the fix conditionally setting `short_chanel_id` in the event such that it's not set in the case? It's still in the event's `previously_failed_channels`, though. Should we similarly add that only conditionally? IIUC, `retry` will eventually be removed.